tracer/src/Datadog.Trace/Util/Http/QueryStringManager.cs

```diff
@@ -12,15 +13,29 @@ namespace Datadog.Trace.Util.Http
     internal class QueryStringManager
     {
         private readonly bool _reportQueryString;
+        private readonly int _maxSizeBeforeObfuscation;
         private readonly Lazy<ObfuscatorBase> _obfuscatorLazy;
 
-        internal QueryStringManager(bool reportQueryString, double timeout, string pattern = null, IDatadogLogger logger = null)
+        internal QueryStringManager(bool reportQueryString, double timeout, int maxSizeBeforeObfuscation, string pattern, IDatadogLogger logger = null)
         {
             _reportQueryString = reportQueryString;
-            pattern ??= Tracer.Instance.Settings.ObfuscationQueryStringRegex;
-            _obfuscatorLazy = new(() => ObfuscatorFactory.GetObfuscator(timeout, pattern, logger, reportQueryString));
+            _maxSizeBeforeObfuscation = maxSizeBeforeObfuscation;
+            _obfuscatorLazy = new(() => ObfuscatorFactory.GetObfuscator(timeout, pattern, logger, _reportQueryString));
         }
 
-        internal string Obfuscate(string queryString) => !_reportQueryString ? string.Empty : _obfuscatorLazy.Value.Obfuscate(queryString);
+        internal string TruncateAndObfuscate(string queryString)
+        {
+            if (!_reportQueryString)
+            {
+                return string.Empty;
+            }
+
+            if (_maxSizeBeforeObfuscation > 0)
+            {
+                queryString = queryString.Substring(0, Math.Min(queryString.Length, _maxSizeBeforeObfuscation));
```

Comment:
I know it _shouldn't_ be, but technically `querystring` could be null, and we'd get a null ref exception. I suggest either adding an `string.IsNullOrEmpty()` for safety or at least adding `#nullable enable`.

Reply:
Thanks. Fixed in 72e7e28d04bd04864b78611fd37a823d18008eba